File: Introduction/Week 4/greedy_algorithms_starter_files/change/change.cpp
#include <iostream>

int get_change(int m) {
  //write your code here
  int n = 0;
  int arr[3];
  
  arr[0] = 10;
  arr[1] = 5;
  arr[2] = 1;

  for(int i = 0; i < 3;i++) {
          n = n + int(m / arr[i]);
          m = m - (int(m / arr[i]) * arr[i]);
  } 
  return n;
}

int main() {
  int m;
  std::cin >> m;
  std::cout << get_change(m) << '\n';
}

File: Introduction/Week 4/greedy_algorithms_starter_files/change/change.py
# Uses python3
import sys

def get_change(m):
    #write your code here
    coin = [10, 5,1]
    count = 0

    for i in coin:
        if m > i:
            count = count + (m // i)
            m = m - (m // i) * i

    return count 

if __name__ == '__main__':
    m = int(input())
    print(get_change(m))
